src/testutils.rs

```diff
@@ -69,15 +67,20 @@ impl Token {
             public_key: from.public.to_bytes().into_val(&self.env),
             signature: from.sign(msg).unwrap().into_val(&self.env),
         });
-        approve(&self.env, &self.contract_id, &auth, &nonce, spender, amount)
+        TokenClient::new(&self.env, &self.contract_id).approve(
```

Comment:
Something you might like to do is move the construction of the client into a helper function on your type, since all the inputs to construction are already on `self.`. So here it would become `self.token_client().approve(`.